test/basicfuns.jl

```diff
@@ -44,6 +44,34 @@
     @test iszero(xlog1py(0 + im * 0, -1 + im * Inf))
 end
 
+@testset "xexpx" begin
+    for z in (0, 0.0, 0f0, false, -Inf)
+        @test iszero(@inferred xexpx(z))
+    end
+    for nan in (NaN16, NaN32, NaN64)
+        @test isnan(@inferred xexpx(nan))
+    end
+    for inf in (Inf16, Inf32, Inf64)
+        @test inf === @inferred xexpx(inf)
+    end
+    for u in (1, true, 1.0, 1f0)
+        @test exp(u) === @inferred xexpx(u)
+    end
+    @test 2exp(2) ≈ @inferred xexpx(2)
+end
+
+@testset "xexpy" begin
+    @test iszero(@inferred xexpy(Inf, -Inf))
+    @test isnan(@inferred xexpy(NaN, -Inf))
+    @test isnan(@inferred xexpy(NaN, 1))
+    @test isnan(@inferred xexpy(0, NaN))
+    @test iszero(@inferred xexpy(0, -Inf))
+    @test iszero(@inferred xexpy(1., -Inf))
+    for x = -3:3, y = -3:3
+        @test x * exp(y) ≈ @inferred xexpy(x, y)
+    end
```

Comment:
Again, maybe it's sufficient to test one/two random numbers, e.g.
```suggestion
    for x in (randn(), randn(Float32)), y in (randn(), randn(Float32))
        correct = x * exp(y)
        @test @inferred(xexpy(x, y)) isa typeof(correct)
        @test xexpy(x, y) ≈ correct
        @test xexpy(x, x) ≈ xexpx(x)
        @test xexpy(y, y) ≈ xexpx(y)
    end
```